docs/dev/testing-hash-join-benchmark.md

```diff
@@ -0,0 +1,78 @@
+# OpenSearch SQL Hash Join Benchmark
+
+## Summary
+
+New hash algorithm introduced memory circuit breaker logic to improve cluster's stability which includes:
+
+1) Rejecting new requests
+2) Terminating ongoing requests with half chance
+3) Retrying remaining requests with binary exponential backoff when memory circuit breaker being triggered
+
+The result summary is as following:
+
+* For cluster stability: the cluster turned from green to yellow due to 4 nodes OOM when using 112 clients sending requests against old hash algorithm. For new hash algorithm, no OOM happened when increasing client's number from 1 to 256.
+* For error rate (Graph 1): there are 3 phases as below
+    1) Before memory circuit breaker being triggered, the error rate of new and old hash is almost same
+    2) Between memory circuit breaker being triggered and OOM, old hash is slightly better than new hash, as new hash starting rejecting new request and terminating some ongoing requests
+    3) After OOM, new hash is much better than old hash
+* For throughput: new hash is roughly 1.4x larger than old hash at peak ( Graph 2 and Graph 3). The new hash throughput is able to stay flat with small drop with increased traffic after peak, while old hash drops to almost 0 dramatically
+* For latency: new hash is roughly 2x faster than old hash ( Graph 4 and Graph 5). The new hash latency increase gradually with increased traffic, while old hash increases quickly to timeout
+
```

Comment:
Maybe add a footnote what is OOM. Out of Memory?

Reply:
Sure, will do. Thanks!

Reply:
https://github.com/opensearch-project/sql/pull/1568/commits/b6f3f1b85199b4689c9032fa02edab2db38f74dc